### ISSUE TITLE
Any guide for training our own .pt?

### ISSUE THREAD

**Issue by tcflying** [vicgalle/stable-diffusion-aesthetic-gradients#10]
Thank you for the great work first, but is there any guide for training our own .pt? eg. how many sample picture should be training one or ? and batch size? Thanks.

**Comment by eeyrw** [comment]
This method do not train model in advance. It trains/finetunes language model CLIP to make your input text embedding similar to the style embedding you created in runtime. So you just use https://github.com/vicgalle/stable-diffusion-aesthetic-gradients/blob/main/scripts/gen_aesthetic_embeddings.py to generate image embedding created by pretrained CLIP image encoder.

**Comment by tcflying** [comment]
thanks a lot of the reply, i got it. exactly, i mean i should prepare how many pictures or batch to create embedding?

**Comment by eeyrw** [comment]
According repo owner's example, 3 images should be sufficient.

**Comment by tcflying** [comment]
> According repo owner's example, 3 images should be sufficient.

thank you guys, as vicgalle mention: fantasy.pt: created from https://huggingface.co/datasets/ChristophSchuhmann/improved_aesthetics_6.5plus by filtering only the images with word "fantasy" in the caption. The top 2000 images by score are selected for the embedding.
flower_plant.pt: created from https://huggingface.co/datasets/ChristophSchuhmann/improved_aesthetics_6.5plus by filtering only the images with word "plant", "flower", "floral", "vegetation" or "garden" in the caption. The top 2000 images by score are selected for the embedding.

he use 2000 pics, Also, how many batch? UM...

**Comment by eeyrw** [comment]
According chapter _Using your own embeddings_, he use 3 [images ](https://github.com/vicgalle/stable-diffusion-aesthetic-gradients/tree/main/reference_images/aivazovsky)as input to create embedding and achieve acceptable performance. But as you said, the author used thousands images to create embedding for actual usage. Maybe much is better. What do you mean of "batch"?
